app/models/manageiq/providers/azure/cloud_manager/deployment.rb

```diff
@@ -0,0 +1,27 @@
+module ManageIQ::Providers::Azure::CloudManager::Deployment
+  extend ActiveSupport::Concern
+
+  def deployment_failed?(deployment)
+    deployment.properties.provisioning_state.casecmp('failed') == 0
+  end
+
+  # Azure deployment does not contain failure reason. The actual reasons exist in the operations.
+  # This method finds the FIRST error message from the operations. It can be used as the failure reason for the stack.
+  # Note: There may be multiple failure reasons.
+  def deployment_failure_reason(deployment_operations)
+    deployment_operations.each do |operation|
+      message = operation_status_message(operation)
+      return message unless message.blank?
+    end
+    nil
+  end
+
+  def operation_status_message(operation)
+    return nil unless operation.properties.respond_to?(:status_message)
+    if operation.properties.status_message.respond_to?(:error)
+      operation.properties.status_message.error.message
+    else
+      operation.properties.status_message.to_s
+    end
+  end
```

Comment:
@bzwei - Can we simplify? here are a couple of suggestions:
1)  Pass ```operation.properties``` into the method instead of just ```operation```
2) I have not tested this, but maybe something along these lines?
    ```    status_message = properties.try(:status_message)```
    ```    return nil unless status_message```
    ```error_message  = status_message.try(:error).try(:message)```
    ```    error_message.nil? ? status_message.to_s : error_message```